collection/collection.go

```diff
@@ -159,6 +165,19 @@ func (c Collection[K, V]) Map(closure func(k K, v V) V) Collection[K, V] {
 	return CollectMap(mappedValues)
 }
 
+func (c Collection[K, V]) Only(keys []K) Collection[K, V] {
+	onlyValues := make(map[K]V)
```

Comment:
I think it _might_ be a good idea to `make` a map with `len(keys)` capacity. Could you add a benchmark to check if it's worth it?

Reply:
By the way, you can also use `makeCollection` for this. Then, you could use `Put` on your for loop and drop the `CollectMap` call on 178.

Reply:
This way:
BenchmarkOnly-12    	 1848312	       649.1 ns/op

With Put and makeCollection
BenchmarkOnly-12    	 1908670	       565.1 ns/op